### PR TITLE
support 'to' option for chardev device with tcp_socket backend

### PR DESCRIPTION
**Pull request by michiboo**
ID: 1792851
Signed-off-by: Micky Yun Chan <michan@redhat.com>

### CODE REVIEW

File: virttest/qemu_devices/qcontainer.py [hunk 1]
```diff
@@ -2255,7 +2255,11 @@ def chardev_define_by_params(self, chardev_id, params, file_name=None):
         if 'socket' in backend:  # tcp_socket & unix_socket
             chardev_param.update(
                 {'server': params.get('chardev_server', 'yes'),
-                 'nowait': params.get('chardev_nowait', 'yes')})
+                 'nowait': params.get('chardev_nowait', 'yes'),
+                 'host': params.get('chardev_host', ''),
+                 'port': params.get('chardev_port', ''),
+                 'to': params.get('chardev_to','')
```
Comment (yanan-fu) [inline virttest/qemu_devices/qcontainer.py]: Only tcp socket have option `to`

Comment (yanan-fu) [reply virttest/qemu_devices/qcontainer.py]: And, `chardev_host` and `chardev_port` value are setted in qemu_vm.py, and used in L2251~L2252

**Comment by yanan-fu** [comment]
Fix the failing please !!!

File: virttest/qemu_devices/qcontainer.py (hunk 1 above)
Comment (michiboo) [reply virttest/qemu_devices/qcontainer.py]: hi in virttest/qemu_vm.py L2251~L2252, it's                 
devs[-1].set_param("if", "pflash")
devs[-1].set_param("format", "raw")
This is the only qemu_vm.py file I can found

Comment (yanan-fu) [reply virttest/qemu_devices/qcontainer.py]: @michiboo     https://github.com/avocado-framework/avocado-vt/blob/e2d51980f53abc63810c2813b256a4c1e4adadb8/virttest/qemu_vm.py#L1771

File: virttest/qemu_devices/qcontainer.py [hunk 2]
```diff
@@ -2252,10 +2252,17 @@ def chardev_define_by_params(self, chardev_id, params, file_name=None):
                  'port': params['chardev_port'],
                  'ipv4': params.get('chardev_ipv4'),
                  'ipv6': params.get('chardev_ipv6')})
+        if 'socket' == backend:
```
Comment (yanan-fu) [inline virttest/qemu_devices/qcontainer.py]:
```
if  'tcp_socket' in backend:
    chardev_param.update({'to': params.get('chardev_to')})
```

File: virttest/qemu_devices/qcontainer.py [hunk 3]
```diff
@@ -2252,10 +2252,16 @@ def chardev_define_by_params(self, chardev_id, params, file_name=None):
                  'port': params['chardev_port'],
                  'ipv4': params.get('chardev_ipv4'),
                  'ipv6': params.get('chardev_ipv6')})
+        if 'tcp_socket' == backend:
```
Comment (yanan-fu) [inline virttest/qemu_devices/qcontainer.py]: prefer `backend == 'tcp_socket'`
```
if backend == 'tcp_socket':
    chardev_param.update('to': params.get('chardev_to')})
```

File: virttest/qemu_devices/qcontainer.py [hunk 4]
```diff
@@ -2252,10 +2252,16 @@ def chardev_define_by_params(self, chardev_id, params, file_name=None):
...
+            chardev_param.update(
+                {
+                'to': params.get('chardev_to','')
+                })
         if 'socket' in backend:  # tcp_socket & unix_socket
             chardev_param.update(
                 {'server': params.get('chardev_server', 'yes'),
-                 'nowait': params.get('chardev_nowait', 'yes')})
+                 'nowait': params.get('chardev_nowait', 'yes')
+                 })
```
Comment (yanan-fu) [inline virttest/qemu_devices/qcontainer.py]: Do not update here.

**Comment by yanan-fu** [comment]
@michiboo   Fix the failing please,and use suitable commit title, we only add "to" now.
`support 'to' option for chardev device with tcp_scoket backend`

**Comment by yanan-fu** [comment]
@luckyh could you help review, thanks!

File: virttest/qemu_devices/qcontainer.py [hunk 5]
```diff
@@ -2252,6 +2252,8 @@ def chardev_define_by_params(self, chardev_id, params, file_name=None):
                  'port': params['chardev_port'],
                  'ipv4': params.get('chardev_ipv4'),
                  'ipv6': params.get('chardev_ipv6')})
+        if backend == 'tcp_socket':
+            chardev_param.update({'to': params.get('chardev_to', '')})
```
Comment (luckyh) [inline virttest/qemu_devices/qcontainer.py]: Suggest removing the default value `''`.

Comment (yanan-fu) [reply virttest/qemu_devices/qcontainer.py]: @luckyh updated, please help review again, thanks!

**Comment by luckyh** [comment]
Thanks all.

**Pull request merged**